emscripten-functions/src/websocket.rs

```diff
@@ -0,0 +1,350 @@
+use emscripten_functions_sys::websocket::*;
+use std::ffi::CString;
+
+// NOTE: Need to add EMCC_CFLAGS="[...] -lwebsocket.js" before cargo build --target=wasm32-unknown-emscripten to prevent javascript linking error
```

Comment:
Oh, so `-lwebsocket.js` needs to be added to the linker flags. It can be added to the `build.rs` like this:
```rust
println!("cargo:rustc-link-arg=-lwebsocket");
```

There would probably need to be an example added to show how to use the websocket API, though it goes beyond the scope of this PR. - I'll probably make one.